app/src/main/AndroidManifest.xml

```diff
@@ -3,6 +3,7 @@
     package="com.example.lhi06.residue">
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.WRITE_INTERNAL_STORAGE"/>
```

Comment:
its hard when im trying to work things out not to do that as i dont really know what anything does when approaching a new subject, but i understand what you mean yes